### PR TITLE
Handle repository events to soft-delete projects from system

### PR DESCRIPTION
**Pull request by matteo95g**
## What does this PR do?

Handles repository events from github, so we can process them and soft-delete projects from our system when those repos are `archived`, `transferred` or `deleted`.

This PR is one part of the full feature.

Resolves https://rootstrap.atlassian.net/jira/software/projects/EM/boards/42?selectedIssue=EM-201 (in part)

### CODE REVIEW

File: spec/services/builders/events/repository_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe Builders::Events::Repository do
```
Comment (martinjaimem) [inline spec/services/builders/events/repository_spec.rb:3]:
```suggestion
describe Builders::Events::Repository do
```

**Review by martinjaimem — APPROVED** [review]
👌

File: app/services/action_handlers/repository.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module ActionHandlers
```
Comment (santib) [inline app/services/action_handlers/repository.rb:1]: Any reason why this file isn't tested?

**Review by santib — APPROVED** [review]
Good job 🙌

File: app/services/action_handlers/repository.rb (hunk 2 above)
Comment (matteo95g) [reply app/services/action_handlers/repository.rb:1]: I added tests for this file in the other PR https://github.com/rootstrap/rs-code-review-metrics/pull/442

**Pull request merged**